storm-client/src/jvm/org/apache/storm/executor/Executor.java

```diff
@@ -124,6 +126,7 @@
     protected int idToTaskBase;
     protected String hostname;
     private static final double msDurationFactor = 1.0 / TimeUnit.MILLISECONDS.toNanos(1);
+    private boolean needToRefreshCreds = true;
```

Comment:
This needs to be `volatile ` to make sure the changes from other threads can be seen immediately

Reply:
Since multiple threads are changing this flag- switching to _AtomicBoolean_.

Reply:
If only simple `get()` and `set()` methods are used in `AtomicBoolean`, it is basically the same as using `violate boolean`. 
https://github.com/AdoptOpenJDK/openjdk-jdk8u/blob/master/jdk/src/share/classes/java/util/concurrent/atomic/AtomicBoolean.java#L85-L87

unless methods like `compareAndSet` and `getAndSet` are required. 

But anyway, using `AtomicBoolean` is easier to understand.